launch_ros/launch_ros/actions/node.py

```diff
@@ -158,6 +182,70 @@ def node_name(self):
             raise RuntimeError("cannot access 'node_name' before executing action")
         return self.__final_node_name
 
+    def _create_params_file_from_dict(self, context, params):
+        with NamedTemporaryFile(mode='w', prefix='launch_params_', delete=False) as h:
+            param_file_path = h.name
+            # TODO(dhood): clean up generated parameter files.
```

Comment:
Is the intention to merge this and follow up with a pull request to clean up the files or do you think we can do without it?

I know we talked about it before and I said it would be ok to not do that for now, but now I'm thinking we should really clean up after ourselves. What are your thoughts?

Reply:
I agree that it's important also, was planning to leave it as followup to simplify reviewing of this PR. The tests are relying on the files being around atm so was going to also have an option to disable the cleanup.